internal/controller/bucket_controller.go

```diff
@@ -140,13 +136,30 @@ func (r BucketReconciler) reconcileNormal(ctx context.Context, objectStorageServ
 		}
 		logger.Info("Bucket access role created")
 	}
+
+	// We update the bucket at the end
+	if err = r.Client.Status().Patch(ctx, bucket, client.MergeFrom(originalBucket)); err != nil {
+		return ctrl.Result{}, errors.WithStack(err)
+	}
 	return ctrl.Result{}, nil
 }
 
 // reconcileDelete deletes the bucket.
 func (r BucketReconciler) reconcileDelete(ctx context.Context, objectStorageService objectstorage.ObjectStorageService, accessRoleService objectstorage.AccessRoleService, bucket *v1alpha1.Bucket) error {
 	logger := log.FromContext(ctx)
 
+	// TODO(quentin) Remove this, this is only to create the role for glippy
+	logger.Info("Creating bucket access role")
+	originalBucket := bucket.DeepCopy()
+	err := accessRoleService.ConfigureRole(ctx, bucket)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	if err = r.Client.Status().Patch(ctx, bucket, client.MergeFrom(originalBucket)); err != nil {
+		return errors.WithStack(err)
+	}
+	logger.Info("Bucket access role created")
+
```

Comment:
I am confused now, why are we creating a role in the delete method ?

Reply:
Sée thé todo above? It's because we have a bucket in deleting state on glippy so it's only for testing but i'll remove it